codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -107,20 +239,85 @@ protected Result beginEncode(HttpResponse headers, CharSequence acceptEncoding)
+     * @param response the response to check.
+     * @return <tt>true</tt> if the response should be compressed, <tt>false</tt> otherwise.
+     */
+    protected boolean isCompressable(HttpResponse response) {
+        CharSequence contentEncoding = response.headers().get(HttpHeaderNames.CONTENT_ENCODING);
+        if (contentEncoding != null && !HttpHeaderValues.IDENTITY.equalsIgnoreCase(contentEncoding)) {
+            // Another encoding is already being applied -> no compression
+            return false;
+        }
+
+        // If compressableContentTypes is null we always turn on compression for backward compatibility
+        if (compressableContentTypes == null) {
+            return true;
+        }
+
+        if (!(response instanceof FullHttpResponse)) {
+            if (!HttpHeaderValues.CHUNKED.equals(response.headers().get(HttpHeaderNames.TRANSFER_ENCODING))) {
+                // We either need a full response or a chunked transfer to turn on compression. If neither of both is
+                // present, we risk that the caller sends a i.e. DefaultFileReqion which would bypass our
+                // encoder completely resulting in an invalid response.
+                return false;
+            }
+        }
+
+        // Check if it is effective to compress the given content (compressing JPEG images wouldn't be a good idea)
+        int contentLength = response.headers().getInt(HttpHeaderNames.CONTENT_LENGTH, 0);
+        CharSequence contentType = response.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        return isCompressionEffective(contentLength, contentType);
+    }
+
+    /**
+     * Determines if it is effective to compress a request with the given content-length and content-type.
+     * <p/>
+     * Only requests which are larger than a certain size and contain reasonable compressable content should
+     * be compressed. If wouldn't make sense to compress a 100 byte JSON response or a JPEG image. The parameters
+     * for this decision are set by <tt>minCompressableContentLength</tt> and <tt>compressableContentTypes</tt>.
+     * <p/>
+     * This method is made public so that other handlers can decide which kind of response to generate. For example
+     * when sending a file it might be better to send it via <code>new HttpChunkedInput(new ChunkedFile(..))</code>
+     * which enables compression rather than via <code>new DefaultFileRegion(...)</code> which permits a zero
+     * copy transfer - but without compression.
+     *
+     * @param contentLength the expected length of the content in bytes or <tt>0</tt> if the length is unknown
+     * @param contentType   the expected type of the content as mime type (e.g. text/xml) or <tt>null</tt> if
+     *                      the type is not yet known
+     * @return <tt>true</tt> if it is considered effective to turn on compression, <tt>false</tt> otherwise
+     */
+    public boolean isCompressionEffective(int contentLength, CharSequence contentType) {
```

Comment:
I knew you'd ask this question, that's why I put the following in the JavaDoc (see below) - If you don't share this intend, I can make it private....:

This method is made public so that other handlers can decide which kind of response to generate. For example when sending a file it might be better to send it via <code>new HttpChunkedInput(new ChunkedFile(..))</code> which enables compression rather than via <code>new DefaultFileRegion(...)</code> which permits a zero copy transfer - but without compression.